### ISSUE TITLE
About the problem " reduce failed to synchronize: device-side assert triggered"

### ISSUE THREAD

**Issue by zrl4836** [eriklindernoren/PyTorch-YOLOv3#229]
PyTorch-YOLOv3/models.py", line 231, in forward
loss_conf_obj = self.bce_loss(pred_conf[obj_mask], tconf[obj_mask])

pred_conf[obj_mask] is tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan], grad_fn=<IndexBackward>)
tconf[obj_mask] is tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
#############################
How can I deal the 'nan' about pred_conf[obj_mask]??

**Comment by LifeIsSoSolong** [comment]
have you solved this problem? I come across same error when training yolov3 using coco dataset.  thanks ~

**Comment by zrl4836** [comment]
Maybe you can add torch.sigmoid.And make sure you have a proper learning rate.I succeeded, but got a bad result.Good luck.

**Comment by zrl4836** [comment]
> have you solved this problem? I come across same error when training yolov3 using coco dataset. thanks ~
Maybe you can add torch.sigmoid.And make sure you have a proper learning rate.I succeeded, but got a bad result.Good luck.

**Comment by LifeIsSoSolong** [comment]
> > have you solved this problem? I come across same error when training yolov3 using coco dataset. thanks ~
> > Maybe you can add torch.sigmoid.And make sure you have a proper learning rate.I succeeded, but got a bad result.Good luck.

original adam lr is 1e-3.  My batch_size=4, and I set lr 1e-4 ,5e-5, 1e-6 . but no use...
Do you training yolov3 using cocodata ? can you share your lr ?

**Comment by LifeIsSoSolong** [comment]
> > have you solved this problem? I come across same error when training yolov3 using coco dataset. thanks ~
> > Maybe you can add torch.sigmoid.And make sure you have a proper learning rate.I succeeded, but got a bad result.Good luck.

the original code has sigmoid(predictions)  to get pded_conf, but I find the value in predictions is "nan", So sigmoid() return nan also...

predictions are computed from images and all computition is conv. So I'm confused Why predictions which as input of yolo layer are nan?

**Comment by LifeIsSoSolong** [comment]
thanks for your relpy~

**Comment by Flova** [comment]
This part of the code has been updated or removed. I, therefore, close this issue now. Feel free to reopen for further discussion.
